### DISCUSSION TITLE
Point-free style for lambdas accessing instance members

### DISCUSSION THREAD

**Discussion by jamesqo** [dotnet/csharplang#1264] [General]
I'm sure there is already a proposal out there for this somewhere. What I have is mind is having the compiler recognize

```cs
IEnumerable<string> names = people.Select(.Name);
```

as shorthand for

```cs
IEnumerable<string> names = people.Select(p => p.Name);
```

Can someone link me to such a proposal? If there isn't already one, I'll expand this one.

**Comment by HaloFour** [comment]
#91 is probably the closest.

**Comment by TonyValenti** [comment]
Good idea. I prefer the syntax for #91.

**Comment by yaakov-h** [comment]
Do you have a proposed solution for handling nested lambdas?

**Comment by DavidArno** [comment]
It seems odd to propose a shorthand that saves typing `p => p`, whilst ignoring the elephant in the room, namely needlessly specifying `IEnumerable<string>`, instead of just using `var`:
```cs
var names = people.Select(p => p.Name); // even shorter than your "shortcut"
```

**Comment by yaakov-h** [comment]
```cs
var names = people.Select(.Name);
```

How short can we go? 😄

**Comment by MkazemAkhgary** [comment]
@yaakov-h 

    var n = p.Select(.N);

**Comment by bondsbw** [comment]
```c#
var names = people select Name;
```

**Comment by jamesqo** [comment]
>  whilst ignoring the elephant in the room, namely needlessly specifying `IEnumerable<string>`

@DavidArno I did that to demonstrate what the return value should be.

> Do you have a proposed solution for handling nested lambdas?

@yaakov-h Like `p => p1 => f(p, p1)`? No, I don't think so. What's the use case for nested lambdas?

**Comment by orthoxerox** [comment]
@bondsbw 

```c#
var names = people.!Name;
```

**Comment by Mafii** [comment]
I personally think the introduction of this/these syntaxes will make the language harder to read.

I don't think it is worth the slightly shorter typing, that a developer has to do. I like the concept, but I don't really think it's worth it.
